opsdroid/connector/matrix/connector.py

```diff
@@ -325,6 +332,8 @@ async def _parse_sync_response(self, response):
 
     async def listen(self):  # pragma: no cover
         """Listen for new messages from the chat service."""
+        if self.connection_failed is True:
+            return
         while True:  # pylint: disable=R1702
```

Comment:
Perhaps instead of having the above check we can do something like this what do you think?

```python
while not self.connection_failed:
```